go/test/endtoend/transaction/twopc/twopc_test.go

```diff
@@ -1449,6 +1449,228 @@ func TestReadTransactionStatus(t *testing.T) {
 	wg.Wait()
 }
 
+// TestVindexes tests that different vindexes work well with two-phase commit.
+func TestVindexes(t *testing.T) {
+	testcases := []struct {
+		name        string
+		initQueries []string
+		testQueries []string
+		logExpected map[string][]string
+	}{
+		{
+			name: "Lookup Single Update",
+			initQueries: []string{
+				"insert into twopc_lookup(id, col, col_unique) values(4, 4, 6)",
+				"insert into twopc_lookup(id, col, col_unique) values(6, 4, 9)",
+				"insert into twopc_lookup(id, col, col_unique) values(9, 4, 4)",
+			},
+			testQueries: []string{
+				"begin",
+				"update twopc_lookup set col = 9 where col_unique = 9",
+				"commit",
+			},
+			logExpected: map[string][]string{
+				"ks.redo_statement:80-": {
+					"insert:[VARCHAR(\"dtid-3\") INT64(1) BLOB(\"delete from lookup where col = 4 and id = 6 and keyspace_id = _binary'`\\\\0\\\\0\\\\0\\\\0\\\\0\\\\0\\\\0' limit 10001\")]",
+					"insert:[VARCHAR(\"dtid-3\") INT64(2) BLOB(\"insert into lookup(col, id, keyspace_id) values (9, 6, _binary'`\\\\0\\\\0\\\\0\\\\0\\\\0\\\\0\\\\0')\")]",
+					"delete:[VARCHAR(\"dtid-3\") INT64(1) BLOB(\"delete from lookup where col = 4 and id = 6 and keyspace_id = _binary'`\\\\0\\\\0\\\\0\\\\0\\\\0\\\\0\\\\0' limit 10001\")]",
+					"delete:[VARCHAR(\"dtid-3\") INT64(2) BLOB(\"insert into lookup(col, id, keyspace_id) values (9, 6, _binary'`\\\\0\\\\0\\\\0\\\\0\\\\0\\\\0\\\\0')\")]",
+				},
+				"ks.twopc_lookup:40-80": {
+					"update:[INT64(6) INT64(9) INT64(9)]",
+				},
+				"ks.lookup:80-": {
+					"delete:[VARCHAR(\"4\") INT64(6) VARBINARY(\"`\\x00\\x00\\x00\\x00\\x00\\x00\\x00\")]",
+					"insert:[VARCHAR(\"9\") INT64(6) VARBINARY(\"`\\x00\\x00\\x00\\x00\\x00\\x00\\x00\")]",
+				},
+			},
+		},
+		{
```

Comment:
nit: test are not checking for dt_state and dt_participants table

Reply:
I know, they didn't feel important in this test.